### PR TITLE
Add checking and error message if materialize features defined on INPUT_CONTEXT

### LINKED ISSUE #767: [EN] Add checking and error message if users want to materialize features that's defined on INPUT_CONTEXT.
### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### Feathr version

0.8

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 20.0)**:
- **Python version**:
- **Spark version, if reporting runtime issue**:

### Describe the problem

We should check if the materialized features are defined on INPUT_CONTEXT, and if it's not, we should give out warning and advise users use the right way to materialize features (maybe link to the corresponding docs)

### Tracking information

_No response_

### Code to reproduce bug

_No response_

### What component(s) does this bug affect?

- [ ] `Python Client`: This is the client users use to interact with most of our API. Mostly written in Python.
- [ ] `Computation Engine`: The computation engine that execute the actual feature join and generation work. Mostly in Scala and Spark.
- [ ] `Feature Registry API`: The frontend API layer supports SQL, Purview(Atlas) as storage. The API layer is in Python(FAST API)
- [ ] `Feature Registry Web UI`: The Web UI for feature registry. Written in React

### PR DESCRIPTION
**Pull request by enya-yx**
## Description

Resolves #767 

## How was this PR tested?

- Tested locally
- Added test case
## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

File: feathr_project/feathr/client.py [hunk 1]
```diff
@@ -619,8 +620,15 @@ def materialize_features(self, settings: MaterializationSettings, execution_conf
             allow_materialize_non_agg_feature: Materializing non-aggregated features (the features without WindowAggTransformation) doesn't output meaningful results so it's by default set to False, but if you really want to materialize non-aggregated features, set this to True.
         """
         feature_list = settings.feature_names
-        if len(feature_list) > 0 and not self._valid_materialize_keys(feature_list):
-            raise RuntimeError(f"Invalid materialization features: {feature_list}, since they have different keys. Currently Feathr only supports materializing features of the same keys.")
+        if len(feature_list) > 0:
+            if 'anchor_list' in dir(self):
+                anchors = [anchor for anchor in self.anchor_list if isinstance(anchor.source, InputContext)]
+                anchor_names = set(feature.name  for anchor in anchors for feature in anchor.features)
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/client.py:626]: `anchor_names` sounds like a name of anchors, but from the context it sounds more like feature names in anchors? We should rename if that's the case

File: feathr_project/feathr/client.py [hunk 2]
```diff
@@ -619,8 +620,15 @@ def materialize_features(self, settings: MaterializationSettings, execution_conf
...
+                for feature in feature_list:
+                    if feature in anchor_names:
+                        raise RuntimeError(f"Invalid materialization feature list: {feature} is defined on INPUT_CONTEXT.")
```
Comment (xiaoyongzhu) [inline feathr_project/feathr/client.py:629]: Materializing features that are defined on INPUT_CONTEXT is not supported. {feature} is defined on INPUT_CONTEXT so you should remove it from the feature list in MaterializationSettings.

**Comment by xiaoyongzhu** [comment]
@enya-yx please refrain from creating branches directly in the main repo. Please use forked repo for development.

**Comment by enya-yx** [comment]
Open a new one using forked repo

**Pull request closed without merging**